contents/Documentation/Effects.playground/Pages/Retrying and repeating effects.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,174 @@
+
+/*:
+ Finally, if we want to keep all intermediate results, we can zip the policy with `Schedule.collectAll()`:
+ */
+let allResults1 = io.repeat(Schedule.collectAll().zipLeft(Schedule.recurs(3)),
+                            onUpdateError: { AnyError.unknown })
+
+// Equivalent to:
+let allResults2 = io.repeat(Schedule.recurs(3).zipRight(Schedule.collectAll()),
+                            onUpdateError: { AnyError.unknown })
+
+/*:
+ ### Repeating an effect until/while it produces a certain value
+ 
+ We can make use of the policies `doWhile` or `doUntil` to repeat an effect while or until its produced result matches a given predicate.
+ */
+
+let repeatWhile = io.repeat(Schedule.doWhile { str in str.isEmpty },
+                            onUpdateError: { AnyError.unknown })
+
+let repeatUntil = io.repeat(Schedule.doUntil { str in str.isEmpty },
+                            onUpdateError: { AnyError.unknown })
+
+/*:
+ However, this may never terminate. In this cases, we can usually add a timeout. We can do this with the `Schedule.duration(t)` policy, that runs an effect for the specified time duration. The examples above can be combined with the `and` combinator to have a timeout, for instance, after 3 seconds:
+ */
+let repeatWhileTimeout = io.repeat(Schedule.doWhile { str in str.isEmpty }
+                                    .and(Schedule.duration(.seconds(3))),
+                                   onUpdateError: { AnyError.unknown })
+
+let repeatUntilTimeout = io.repeat(Schedule.doUntil { str in str.isEmpty }
+                                    .and(Schedule.duration(.seconds(3))),
+                                   onUpdateError: { AnyError.unknown })
+
+/*:
+ ### Exponential backoff retries
+ 
+ A common algorithm to retry effectful operations, as network requests, is the [exponential backoff algorithm](https://en.wikipedia.org/wiki/Exponential_backoff). There is a scheduling policy that implements this algorithm and can be used as:
+ */
+let exponential = io.retry(Schedule.exponential(.milliseconds(250)))
+
+/*:
+ As more iterations of this algorithm are performed, the waiting time becomes larger and larger. We may want to use it for the initial retries, and then switch to evenly spaced retries. This can be achieved with the `Schedule.spaced(t)` policy and the `or` combinator. This will keep retrying as long as either of them want to keep retrying, using the minimum of both delays between iterations:
+ */
+let exponentialOrFixed = io.retry(
+    Schedule.exponential(.milliseconds(250)).or(Schedule.spaced(.seconds(3)))
+)
+/*:
+ This will attempt to retry the effect with the exponential backoff algorithm, waiting at most 3 seconds between each attempt. Nevertheless, this will keep running forever. We can limit it in time, using the `Schedule.duration(t)` policy, to set a timeout, or limit the number of iterations with the `Schedule.recurs(n)` policy. For instance, we can limit it to 10 attempts like:
```

Comment:
I think we could split this sentence in two, one with the `.and(Schedule.recurs(10)` example and another for `.and(Schedule.duration(.seconds(30))`